### PR TITLE
am57xx: initial support for TI AM57xx

### PR DESCRIPTION
**Pull request by igoropaniuk**
Also related minor PRs:
https://github.com/linaro-swg/gen_rootfs/pull/22
https://github.com/OP-TEE/optee_os/pull/1494
https://github.com/OP-TEE/manifest/pull/51/files

`Signed-off-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

### CODE REVIEW

**Comment by igoropaniuk** [comment]
@glneo Please have a look at this PR

File: ti/fitImage-am57xx.its [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			loadables = "optee";
+		};
+
+		am57xx-evm-reva3.dtb {
+			description = "Linux kernel, FDT blob, OPTEE OS";
+			kernel = "kernel@1";
+			fdt = "am57xx-evm-reva3.dtb";
+			loadables = "optee";
+		};
+
```
Comment (glneo) [inline ti/fitImage-am57xx.its]: Extra line.

File: ti/fitImage-am57xx.its [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			compression = "none";
+		};
+
+		optee {
+			description = "OPTEE OS Image";
+			data = /incbin/("tee.bin.sec");
+			type = "tee";
+			arch = "arm";
+			compression = "none";
+			load = <0xbdb000e4>;
```
Comment (glneo) [inline ti/fitImage-am57xx.its]: The final load location for OP-TEE is firewalled, so U-Boot cannot load OP-TEE here, we load OP-TEE to an arbitrary location and then let the PPA (the initial secure software that we can call into) do the relocation to behind the firewall before passing control over. So this load line needs to be dropped.

File: README.md [hunk 3]
```diff
@@ -104,6 +105,7 @@ Here is a list of manifests for the devices currently supported.
 | ARM Juno board| `juno.xml` | `juno_stable.xml` |
 | Raspberry Pi 3 | `rpi3.xml` | `rpi3_stable.xml` |
 | DRA7xx | `dra7xx.xml` | `dra7xx_stable.xml` |
+| AM57xx | `am57xx.xml` | N/A |
```
Comment (glneo) [inline README.md]: An am57xx_stable.xml will be generated when the dust settles, lets add the reference here so we don't forget later. No harm in that I think.

File: ti/fitImage-am57xx.its (hunk 2 above)
Comment (igoropaniuk) [inline ti/fitImage-am57xx.its]: BTW: that's why I was receiving a security violation error, which I noticed in the last email, right?

Comment (glneo) [inline ti/fitImage-am57xx.its]: Probably, touching firewalled memory, even pre-fetches and speculative access, will throw exceptions, so trying to load a binary to one will certainly fail spectacularly.

File: README.md (hunk 3 above)
Comment (igoropaniuk) [inline README.md]: fixed

File: ti/fitImage-am57xx.its (hunk 1 above)
Comment (igoropaniuk) [inline ti/fitImage-am57xx.its]: fixed

File: ti/fitImage-am57xx.its (hunk 2 above)
Comment (igoropaniuk) [inline ti/fitImage-am57xx.its]: thanks for this info. Fixed!

**Comment by glneo** [comment]
```Reviewed-by:  Andrew F. Davis <afd@ti.com>```

**Comment by igoropaniuk** [comment]
Applied tags & rebased

**Comment by igoropaniuk** [comment]
QEMUv8 build has false negative results, it fails on fetching deb packages:
```
E: Failed to fetch http://us-central1.gce.archive.ubuntu.com/ubuntu/pool/main/libc/libcap2/libcap-dev_2.24-0ubuntu2_amd64.deb  503  Service Unavailable [IP: 104.197.134.158 80]
```

**Comment by igoropaniuk** [comment]
@jforissier do I need additional R-b tags to get this PR merged?

**Comment by jforissier** [comment]
@igoropaniuk no, that's fine, thanks.

**Pull request merged**